server/src/lib/routes/verify/get_session_cookie.ts

```diff
@@ -74,6 +72,9 @@ export default function (req: Express.Request, res: Express.Response,
     vars.logger.debug(req, "domain=%s, request_uri=%s, user=%s, groups=%s", domain,
       originalUri, username, groups.join(","));
 
+    if (authSession.whitelisted > WhitelistValue.NOT_WHITELISTED)
+      return resolve();
```

Comment:
It's not absolutely required, as any fulfilled or rejected promise will have already been settled so likely any further rejects will have no effect, however, this is not good practice. Once the promise has been resolved or rejected you should almost *always* be returning (or return the invoked function), otherwise the function will continue.